hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -199,6 +199,12 @@ private static void setupCleaningOptions(Configuration conf) {
       conf.setInteger(FlinkOptions.ARCHIVE_MIN_COMMITS, commitsToRetain + 10);
       conf.setInteger(FlinkOptions.ARCHIVE_MAX_COMMITS, commitsToRetain + 20);
     }
+    if (conf.getBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED)
```

Comment:
why not just respect the config user-defined? this change the config without noticing anyone

Reply:
I have decided that if the option value is as the default.

Reply:
IMO, @garyli1019 's opinion is reasonable. @danny0405 I know what's your purpose. Can we define an explicit config option for this? So that users can know this detail information.

Reply:
We have respected the user specified option, the value is rewritten only when the default value is set.

Reply:
Yes, but the user may not know it would be rewritten under some condition like this?

Reply:
Why the user needs to know that ? They can set up the target IO by themself if they want BTW.

Reply:
The rewritten value is a constant value. Why not provide an `SYNC_COMPACTION_TARGET_IO` option and make it to be `500GB`?

Reply:
There is already one, the default value is 5GB for online compaction, this patch fix the offline  compaction with default value as 500GB.

Reply:
sorry, a bit late on this. I don't have a strong objection here but slightly incline to not turning config inside the code.
My point is that tweaking config is ok if the user wrote their own connector or wanted to reduce the overhead of having too many configs in their app. I do this in my app as well. But this is the official connector of Hudi, so I think we should keep the config as how it is.